crates/sui/src/validator_commands.rs

```diff
@@ -357,16 +622,311 @@ impl Debug for SuiValidatorCommandResponse {
+            if json {
+                println!("{}", serde_json::to_string_pretty(&info)?);
+            } else {
+                println!("{:#?}", info);
+            }
         }
     }
+    Ok(())
+}
+
+async fn get_pending_candidate_summary(
+    validator_address: SuiAddress,
+    sui_client: &SuiClient,
+    pending_active_validators_id: ObjectID,
+) -> anyhow::Result<Option<ValidatorV1>> {
+    let pending_validators = sui_client
+        .read_api()
+        .get_dynamic_fields(pending_active_validators_id, None, None)
+        .await?
+        .data
+        .into_iter()
+        .map(|dyi| dyi.object_id)
+        .collect::<Vec<_>>();
+    let resps = sui_client
+        .read_api()
+        .multi_get_object_with_options(
+            pending_validators,
+            SuiObjectDataOptions::default().with_bcs(),
+        )
+        .await?;
+    for resp in resps {
+        let object_id = resp.object_id();
+        let bcs = resp.move_object_bcs().ok_or_else(|| {
+            anyhow::anyhow!(
+                "Object {} does not exist or does not return bcs bytes",
+                object_id
+            )
+        })?;
+        let val = bcs::from_bytes::<ValidatorV1>(bcs).map_err(|e| {
+            anyhow::anyhow!(
+                "Can't convert bcs bytes of object {} to ValidatorV1: {}",
+                object_id,
+                e,
+            )
+        })?;
+        if val.verified_metadata().sui_address == validator_address {
+            return Ok(Some(val));
+        }
+    }
+    Ok(None)
+}
+
+#[derive(Subcommand)]
+#[clap(rename_all = "kebab-case")]
+pub enum MetadataUpdate {
+    /// Update name. Effectuate immediately.
+    Name { name: String },
+    /// Update description. Effectuate immediately.
+    Description { description: String },
+    /// Update Image URL. Effectuate immediately.
+    ImageUrl { image_url: String },
+    /// Update Project URL. Effectuate immediately.
+    ProjectUrl { project_url: String },
+    /// Update Network Address. Effectuate from next epoch.
+    NetworkAddress { network_address: Multiaddr },
+    /// Update Primary Address. Effectuate from next epoch.
+    PrimaryAddress { primary_address: Multiaddr },
+    /// Update Worker Address. Effectuate from next epoch.
+    WorkerAddress { worker_address: Multiaddr },
+    /// Update P2P Address. Effectuate from next epoch.
+    P2pAddress { p2p_address: Multiaddr },
+    /// Update Network Public Key. Effectuate from next epoch.
+    NetworkPubKey {
+        #[clap(name = "network-key-path")]
+        file: PathBuf,
+    },
+    /// Update Worker Public Key. Effectuate from next epoch.
+    WorkerPubKey {
+        #[clap(name = "worker-key-path")]
+        file: PathBuf,
+    },
+    /// Update Protocol Public Key and Proof and Possession. Effectuate from next epoch.
+    ProtocolPubKey {
+        #[clap(name = "protocol-key-path")]
+        file: PathBuf,
+    },
+}
```

Comment:
Does this require someone to batch update everything in one go or can you only update 1 item?